contracts/Joint.sol

```diff
@@ -159,85 +187,288 @@ contract Joint {
+                startingB,
+                startingA,
+                reserveB,
+                reserveA,
+                10**uint256(IERC20Extended(tokenB).decimals())
+            );
+        }
     }
 
-    function setWETH(address _weth) external onlyGov {
-        WETH = _weth;
+    function _calculateSellToBalance(
+        uint256 current0,
+        uint256 current1,
+        uint256 starting0,
+        uint256 starting1,
+        uint256 reserve0,
+        uint256 reserve1,
+        uint256 precision
+    ) internal pure returns (uint256 _sellAmount) {
+        uint256 numerator =
+            current0.sub(starting0.mul(current1).div(starting1)).mul(precision);
+        uint256 denominator;
+        uint256 exchangeRate;
+
+        // First time to approximate
+        exchangeRate = UniswapV2Library.getAmountOut(
+            precision,
+            reserve0,
+            reserve1
+        );
+        denominator = precision + starting0.mul(exchangeRate).div(starting1);
+        _sellAmount = numerator.div(denominator);
+
+        // Second time to account for slippage
+        exchangeRate = UniswapV2Library
+            .getAmountOut(_sellAmount, reserve0, reserve1)
+            .mul(precision)
+            .div(_sellAmount);
+        denominator = precision + starting0.mul(exchangeRate).div(starting1);
+        _sellAmount = numerator.div(denominator);
+    }
+
+    function getRatios(
+        uint256 currentA,
+        uint256 currentB,
+        uint256 startingA,
+        uint256 startingB
+    ) internal pure returns (uint256 _a, uint256 _b) {
+        _a = currentA.mul(RATIO_PRECISION).div(startingA);
+        _b = currentB.mul(RATIO_PRECISION).div(startingB);
+    }
+
+    function getReserves()
+        public
+        view
+        returns (uint256 reserveA, uint256 reserveB)
+    {
+        if (tokenA == pair.token0()) {
+            (reserveA, reserveB, ) = pair.getReserves();
+        } else {
+            (reserveB, reserveA, ) = pair.getReserves();
+        }
+    }
+
+    function createLP()
+        internal
+        returns (
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        // **WARNING**: This call is sandwichable, care should be taken
+        //              to always execute with a private relay
+        return
+            IUniswapV2Router02(router).addLiquidity(
+                tokenA,
+                tokenB,
+                balanceOfA(),
+                balanceOfB(),
+                0,
+                0,
+                address(this),
+                now
+            );
     }
 
     function findSwapTo(address token) internal view returns (address) {
         if (tokenA == token) {
             return tokenB;
         } else if (tokenB == token) {
             return tokenA;
+        } else if (reward == token) {
+            if (tokenA == WETH || tokenB == WETH) {
+                return WETH;
+            }
+            return tokenA;
```

Comment:
More about the swapping to both A and B evenly. Should this always be swapping to A?

Reply:
This is the fallback case for now when on side of the LP isn't weth or the reward token. Maybe this could be configurable or we come up with an alternative method once we have a need for it.